File: 2441/LargestPositiveIntegerThatExistsWithItsNegative.java
/*
https://leetcode.com/problems/largest-positive-integer-that-exists-with-its-negative/description/?envType=daily-question&envId=2024-05-02
*/

class Solution {

    public boolean customSearch(int[] nums, int val){
        int left=0, right=nums.length-1;
        while(left<=right){
            if(nums[left++]==-val) return true;
            else if(nums[right--]==-val) return true;
        }
        return false;
    }

    public int findMaxK(int[] nums) {
        Arrays.sort(nums);
        int sz=nums.length;
        for(int i=sz-1; i>=0; --i){
            int curr=nums[i];
            if(curr>=0){
                boolean flag=customSearch(nums, curr);
                if(flag) return curr;
            }
        }
        return -1;
    }
}

File: 2/2000s/24s/2441/LargestPositiveIntegerThatExistsWithItsNegative.ts
/*
https://leetcode.com/problems/largest-positive-integer-that-exists-with-its-negative/description/?envType=daily-question&envId=2024-05-02
*/

function customSearch(nums:number[], val:number):boolean{
    let left:number=0, right:number=nums.length-1;
    while(left<=right){
        if(nums[left++]===-val) return true;
        else if(nums[right--]===-val) return true;
    }
    return false;
}

function findMaxK(nums: number[]): number {
    nums.sort((A, B)=>(A-B));
    let sz:number=nums.length;
    for(let i:number=sz-1; i>=0; --i){
        let curr:number=nums[i];
        if(curr>=0){
            let flag:boolean=customSearch(nums, curr);
            if(flag===true) return curr;
        }
    }
    return -1;
};